### ISSUE TITLE
Bypass IsDirty on a Property

### ISSUE THREAD

**Issue by CrossSlide** [MarimerLLC/cslaforum#478]
I have a BusinessList with child objects. My child object has a property "Track" that is only used at runtime and not persisted. I need my BusinessList to respond to the OnChildChanged event when Track is changed. But, when Track is changed I don't want the child's persistence state to be changed. The child should remain Old or Dirty regardless of the change to Track. How would I implement this?

**Comment by hurcane** [comment]
Could you define a custom event in the child class and write the business list logic to respond to the custom event?

**Comment by CrossSlide** [comment]
Yes, I think I could do that but I just wanted to make sure there wasn't a "proper" way to do it within the framework. :-) 

Thanks @hurcane

**Comment by hurcane** [comment]
As I think about it some more, I think you could use OnPropertyChanged directly, without having to define a new event. The normal property change workflow includes a call MarkDirty, which I presume you want to avoid. It also checks rules, which I presume you want to avoid. And it calls OnPropertyChanged. This call is what bubbles up to the OnChildChanged event in the list.

**Comment by CrossSlide** [comment]
So are you saying to override the OnPropertyChanged of the child and not call it's base, then call OnChildChanged?

**Comment by hurcane** [comment]
Inside the setter of the Track property, call OnPropertyChanged. That will not cause rules to run and it will not cause MarkDirty to be called. In the list, you will handle it the way you were intending to handle it.

**Comment by rockfordlhotka** [comment]
Also, you wouldn't implement this as a managed property, just as an old-fashioned standard property, where you manually call `OnPropertyChanged` in the setter. 

Additionally, if you are using the BinaryFormatter or NDCS for serialization, make sure you mark the backing field for the property as `NonSerialized`.

**Comment by CrossSlide** [comment]
Great, I'll try that! Thanks @hurcane @rockfordlhotka
